# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalChild.cpp
# Commit: 5686e9cc48b2
# Full Hash: 5686e9cc48b26a8ec550f94091d5b8b4ed840f7d
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1579992
# File Overlap Count: 1
# Description:
#   Bug 1579992 - Load pages into new webCOOP+COEP process type r=nika
#   
#   * This patch makes pages with the `OPENER_POLICY_SAME_ORIGIN_EMBEDDER_POLICY_REQUIRE_CORP` policy load into a special `webCOOP+COEP={pageOrigin}` remote type.
#   * Adds `E10SUtils.WEB_REMOTE_COOP_COEP_TYPE_PREFIX="webCOOP+COEP="`
#   * When a COOP process switch occurs and the target page doesn't have this policy, we pass a `preferredRemoteType="web"` into `E10SUtils.getRemoteTypeForPrincipal` ensuring that we correctly get a different `remoteType`
# ==============================================================================

diff -r fd0d2f340380 -r 5686e9cc48b2 dom/ipc/WindowGlobalChild.cpp
--- a/dom/ipc/WindowGlobalChild.cpp	Mon Oct 21 16:34:43 2019 +0000
+++ b/dom/ipc/WindowGlobalChild.cpp	Mon Oct 21 16:56:00 2019 +0000
@@ -80,7 +80,7 @@
   if (httpChan &&
       loadInfo->GetExternalContentPolicyType() ==
           nsIContentPolicy::TYPE_DOCUMENT &&
-      NS_SUCCEEDED(httpChan->GetCrossOriginOpenerPolicy(
+      NS_SUCCEEDED(httpChan->ComputeCrossOriginOpenerPolicy(
           nsILoadInfo::OPENER_POLICY_NULL, &policy))) {
     bc->SetOpenerPolicy(policy);
   }